src/mjolnir/pbfgraphparser.cc

```diff
@@ -1030,6 +1037,45 @@ struct graph_callback : public OSMPBF::Callback {
       }
     }
 
+    if (use_direction_on_ways_) {
+      if (!direction.empty()) {
+        if (!ref.empty())
+          w.set_ref_index(osmdata_.name_offset_map.index(ref));
+      } else {
+        if (!ref.empty()) {
+          std::vector<std::string> refs = GetTagTokens(ref);
+          std::vector<std::string> directions = GetTagTokens(direction);
+
+          if (refs.size() == directions.size()) {
+            for (uint32_t i = 0; i < refs.size(); i++) {
+              if (!refs.at(i).empty() && !directions.at(i).empty())
+                w.set_ref_index(osmdata_.name_offset_map.index(refs.at(i) + " " + directions.at(i)));
+            }
+          } else w.set_ref_index(osmdata_.name_offset_map.index(ref));
+        }
+      }
+    }
+
+    if (use_direction_on_ways_) {
```

Comment:
Same comment as before, these could be collapsed into a single stmt (less verbose).